nodes/core/io/21-httpin.js

```diff
@@ -232,6 +250,15 @@ module.exports = function(RED) {
                     msg.payload += chunk;
                 });
                 res.on('end',function() {
+                    if(RED.settings.metricsOn && RED.settings.metricsOn === true) {
+                        if(preRequestTimestamp) {
+                            var elapsedTimeDiff = process.hrtime(preRequestTimestamp);
+                            node.metric(msg, "request.duration.nanos", (elapsedTimeDiff[0] * 1e9 + elapsedTimeDiff[1]));
```

Comment:
Are you assuming that Node.metric will prepend "node.{type}." to the event name you are passing here? I think you are, but that isn't what the code does today. I have pulled Helen's commit locally and making some updates before pushing to master.

I had envisaged the node emitting a single metric which contains a number of properties in the metrics object (which is why we didn't include support for a single primitive value). So in this instance, there would be a single call to metric like:

    node.metric(msg,"request",{duration: ..., responseSize: .... })

rather than lots of individual metric events for each individual value.